code/lib/cli/src/scaffold-new-project.ts

```diff
@@ -0,0 +1,176 @@
+import boxen from 'boxen';
+import chalk from 'chalk';
+import prompts from 'prompts';
+import dedent from 'ts-dedent';
+import { logger } from '@storybook/node-logger';
+
+import type { PackageManagerName } from './js-package-manager';
+import { exec } from './utils';
+
+type CoercedPackageManagerName = 'npm' | 'yarn' | 'pnpm';
+
+interface SupportedProject {
+  displayName: {
+    type: string;
+    builder?: string;
+    language: string;
+  };
+  createScript: Record<CoercedPackageManagerName, string>;
+}
+
+/**
+ * The supported projects.
+ */
+const SUPPORTED_PROJECTS: Record<string, SupportedProject> = {
+  'react-vite-ts': {
+    displayName: {
+      type: 'React',
+      builder: 'Vite',
+      language: 'TS',
+    },
+    createScript: {
+      npm: 'npm create vite@latest --yes . -- --template react-ts',
+      yarn: 'yarn create vite@latest --yes . --template react-ts',
+      pnpm: 'pnpm create vite@latest --yes . --template react-ts && pnpm i --prefer-offline',
+    },
+  },
+  'nextjs-ts': {
+    displayName: {
+      type: 'Next.js',
+      language: 'TS',
+    },
+    createScript: {
+      npm: 'npm create next-app . -- --typescript --use-npm --eslint --tailwind --app --import-alias="@/*" --src-dir',
```

Comment:
Do we want `--no-app` here because Storybook doesn't support RSC yet?

Reply:
Good catch! :D updated!